lib/ipfs/IpfsStorage.ts

```diff
@@ -45,32 +45,20 @@ export default class IpfsStorage {
     // If we hit error attempting to fetch the content metadata, return not-found.
     let contentMetadata = undefined;
     try {
-      contentMetadata = await (this.node as any).object.stat(hash);
+      contentMetadata = await this.node.object.stat(hash);
     } catch (error) {
       console.info(error);
       return { code: FetchResultCode.NotFound };
     }
 
-    // If content size cannot be found, return not-found.
-    if (contentMetadata === undefined || contentMetadata.CumulativeSize === undefined) {
-      return { code: FetchResultCode.NotFound };
-    }
-
-    // NOTE: IPFS API does not have an API for finding out only the content size,
-    // IPFS API only provides the "cumulative size" that includes the additional metadata on how the Merkle DAG is formed.
-    // So here we account for the additional space used by performing a more lenient max size check by 10%,
-    // we will track the exact conent size of content later when we fetch the content if the content passes this size check.
-    const adjustedMaxSize = maxSizeInBytes * 1.1;
-    if (contentMetadata.CumulativeSize > adjustedMaxSize) {
-      console.info(`Cumulative size of ${contentMetadata.CumulativeSize} bytes is greater than the ${adjustedMaxSize} cumulative size limit.`);
+    if (contentMetadata.DataSize > maxSizeInBytes) {
+      console.info(`Size of ${contentMetadata.DataSize} bytes is greater than the ${maxSizeInBytes} data size limit.`);
```

Comment:
From what I can tell reading the tests for stat, it will always return a value, but I could be wrong. I will add back the check just to be safe. I think it is good to have and not assume.